File: sdk/mobile/ios/HjtpxCaptchaSDK/HjtpxCaptchaSDK/CaptchaConfig.swift
import Foundation

public class CaptchaConfig {
    public static var shared = CaptchaConfig()

    public var captchaWidth: Int = 320
    public var captchaHeight: Int = 200
    public var enableHapticFeedback: Bool = true
    public var enableSoundEffect: Bool = false
    public var sliderTrackHeight: Int = 4
    public var sliderThumbSize: Int = 50
    public var timeout: TimeInterval = 30.0

    private init() {}

    public func reset() {
        captchaWidth = 320
        captchaHeight = 200
        enableHapticFeedback = true
        enableSoundEffect = false
        sliderTrackHeight = 4
        sliderThumbSize = 50
        timeout = 30.0
    }
}

File: sdk/mobile/flutter/lib/src/captcha_config.dart
class CaptchaConfig {
  int captchaWidth;
  int captchaHeight;
  bool enableHapticFeedback;
  bool enableSoundEffect;
  int sliderTrackHeight;
  int sliderThumbSize;
  int timeout;

  CaptchaConfig({
    this.captchaWidth = 320,
    this.captchaHeight = 200,
    this.enableHapticFeedback = true,
    this.enableSoundEffect = false,
    this.sliderTrackHeight = 4,
    this.sliderThumbSize = 50,
    this.timeout = 30,
  });

  CaptchaConfig copyWith({
    int? captchaWidth,
    int? captchaHeight,
    bool? enableHapticFeedback,
    bool? enableSoundEffect,
    int? sliderTrackHeight,
    int? sliderThumbSize,
    int? timeout,
  }) {
    return CaptchaConfig(
      captchaWidth: captchaWidth ?? this.captchaWidth,
      captchaHeight: captchaHeight ?? this.captchaHeight,
      enableHapticFeedback: enableHapticFeedback ?? this.enableHapticFeedback,
      enableSoundEffect: enableSoundEffect ?? this.enableSoundEffect,
      sliderTrackHeight: sliderTrackHeight ?? this.sliderTrackHeight,
      sliderThumbSize: sliderThumbSize ?? this.sliderThumbSize,
      timeout: timeout ?? this.timeout,
    );
  }
}
